checks/evaluation/permissions/GitHubWorkflowPermissionsStepsNoWrite.yml

```diff
@@ -0,0 +1,18 @@
+short: Checks that GitHub workflows do not have steps with dangerous write permissions
+desc: This rule checks that GitHub workflows do not have steps with dangerous write permissions
+motivation: >
+  Even with permissions default set to read, some scopes having write permissions brings vulnerabilities to the project,
+  such as commiting unreviewed code, remove pre-submit checks to introduce a bug, etc.
+  For more information about the scopes and the vulnerabilities involved, see https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions.
```

Comment:
if we use this as a source of truth to generate checks.md, this may be a circular reference.

Reply:
Good point. No plan yet to use these to generate checks.md yet. I'll keep this in mind when when we get to this part (in a later PR)